DMCompiler/DM/Visitors/DMObjectBuilder.cs

```diff
@@ -72,11 +72,48 @@ class DMObjectBuilder {
 
             _currentObject = DMObjectTree.GetDMObject(varDefinition.ObjectPath);
 
-            if (varDefinition.IsGlobal) {
-                variable = _currentObject.CreateGlobalVariable(varDefinition.Type, varDefinition.Name, varDefinition.IsConst);
-            } else {
-                variable = new DMVariable(varDefinition.Type, varDefinition.Name, false, varDefinition.IsConst);
-                _currentObject.Variables[variable.Name] = variable;
+            //DMObjects store two bundles of variables; the statics in GlobalVariables and the non-statics in Variables.
+            if (varDefinition.IsStatic) { // static
+
+                //make sure this static doesn't already exist first
+                if(DoesOverrideGlobalVars(varDefinition)) // Some snowflake behaviour for global.vars
+                {
+                    DMCompiler.Error(new CompilerError(varDefinition.Location, "Duplicate definition of global.vars"));
+                    //We can't salvage any part of this definiton, since global.vars doesn't technically even exist, so lets just return
+                    _currentObject = oldObject; //TODO: Does C# have deferred things, like with destructor fanciness or something?
+                    return;
+                }
+                if (_currentObject.HasGlobalVariable(varDefinition.Name))
```

Comment:
These checks for existing local/global variables are basically copy-pasted below. This could be moved outside, before the `if (varDefinition.IsStatic)`.

Reply:
You squash my commits _AND_ you reduce my linecount bloat? What clout can I even have now!! 😭

Reply:
Closed issues clout